effect/ref.py

```diff
@@ -0,0 +1,60 @@
+from characteristic import attributes
+
+from ._base import Effect
+from ._dispatcher import TypeDispatcher
+from ._sync import sync_performer
+
+
+class ERef(object):
+    """
+    An effectful mutable variable.
+
+    Compare to Haskell's ``IORef`` or Clojure's ``atom``.
```

Comment:
It does make more sense, but I guess the main reason I was confused was I didn't understand what the difference between this and just mutating a variable was, so I thought I was missing something.

I'm not sure if this would fit better in this docstring or in a wiki, or howto, or some other documentation, but would it make sense to explain that this is basically that equivalent of python variable mutation, but how and why it's better?  The discussion below was fairly enlightening - I would have just had a callback that mutated some object or something, rather than use this effect with a transformer, and it might be useful to discuss why this pattern is better.

:)  Probably super obvious to you, less so to me.